blackboxopt/optimizers/testing.py

```diff
@@ -218,9 +237,48 @@ def raises_evaluation_error_when_reporting_unknown_objective(
     return True
 
 
+def respects_fixed_parameter(
+    optimizer_class: Union[
+        Type[SingleObjectiveOptimizer], Type[MultiObjectiveOptimizer]
+    ],
+    optimizer_kwargs: dict,
+):
+    """Check if optimizer's generated evaluation specifications contain the values
+    a parameter in the search space was fixed to.
+
+    Args:
+        optimizer_class: Optimizer to test.
+        optimizer_kwargs: Expected to contain additional arguments for initializing
+            the optimizer. (`search_space` and `objective(s)` are set automatically
+            by the test.)
+
+    Returns:
+        `True` if the test is passed.
+    """
+    space = ps.ParameterSpace()
+    space.add(ps.ContinuousParameter("my_fixed_param", (-10.0, 200.0)))
+
+    fixed_value = 1.0
+    space.fix(my_fixed_param=1.0)
+
+    for _ in range(5):
```

Comment:
Just out of curiosity, why repeat this 5 times? Should it not be deterministic?

Reply:
There is a small chance that a non-fixed parameter would sample exactly 1.0, and while 5 is certainly arbitrary, I'd still vote for trying more than once.